apps/antalmanac/src/components/RightPane/SectionTable/SectionTable.tsx

```diff
@@ -10,109 +12,122 @@ import {
-    scheduleNoteContainer: {},
-};
-
-const tableHeaderColumns: Record<SectionTableColumn, string> = {
-    sectionCode: 'Code',
-    sectionDetails: 'Type',
-    instructors: 'Instructors',
-    dayAndTime: 'Times',
-    location: 'Places',
-    sectionEnrollment: 'Enrollment',
-    restrictions: 'Restr',
-    status: 'Status',
 };
+const tableHeaderColumnEntries = Object.entries(tableHeaderColumns);
 
-const SectionTable = (props: SectionTableProps) => {
-    const { classes, courseDetails, term, allowHighlight, scheduleNames, analyticsCategory } = props;
-    const courseId = courseDetails.deptCode.replaceAll(' ', '') + courseDetails.courseNumber;
-    const encodedDept = encodeURIComponent(courseDetails.deptCode);
-    const isMobileScreen = useMediaQuery(`(max-width: ${MOBILE_BREAKPOINT}`);
+interface EnrollmentColumnHeaderProps {
+    label: string;
+}
 
-    const [activeColumns, setActiveColumns] = useState(RightPaneStore.getActiveColumns());
+function EnrollmentColumnHeader(props: EnrollmentColumnHeaderProps) {
+    const isMobileScreen = useMediaQuery(`(max-width: ${MOBILE_BREAKPOINT})`);
 
-    const handleColumnChange = useCallback(
-        (newActiveColumns: SectionTableColumn[]) => {
-            setActiveColumns(newActiveColumns);
-        },
-        [setActiveColumns]
+    return (
+        <Box display="flex">
+            {props.label}
+            {!isMobileScreen && (
+                <Tooltip
+                    title={
+                        <Typography>
+                            Enrolled/Capacity
+                            <br />
+                            Waitlist
+                            <br />
+                            New-Only Reserved
+                        </Typography>
+                    }
+                >
+                    <Help fontSize="small" />
+                </Tooltip>
+            )}
+        </Box>
     );
+}
+
+function SectionTable(props: SectionTableProps) {
+    const { courseDetails, term, allowHighlight, scheduleNames, analyticsCategory } = props;
+
+    const { selectedColumns, getActiveColumns } = useColumnStore();
+
+    const isMobileScreen = useMediaQuery(`(max-width: ${MOBILE_BREAKPOINT})`);
 
-    useEffect(() => {
-        RightPaneStore.on('columnChange', handleColumnChange);
+    const courseId = useMemo(() => {
+        return courseDetails.deptCode.replaceAll(' ', '') + courseDetails.courseNumber;
+    }, [courseDetails.deptCode, courseDetails.courseNumber]);
 
-        return () => {
-            RightPaneStore.removeListener('columnChange', handleColumnChange);
-        };
-    }, [handleColumnChange]);
+    const encodedDept = useMemo(() => {
+        return encodeURIComponent(courseDetails.deptCode);
+    }, [courseDetails.deptCode]);
+
+    /**
+     * Limit table width to force side scrolling.
+     */
+    const tableMinWidth = useMemo(() => {
+        const width = isMobileScreen ? 600 : 780;
+        const numActiveColumns = getActiveColumns().length;
```

Comment:
Because `selectedColumns` is being memoized it would be best to use it directly as part of the calculation (i.e. since it's guaranteed to be affecting when this is calculated). And if necessary, the column store should be refactored to accommodate this, i.e. by setting the `selectedColumns` to the desired structure directly, and/or creating separate data structures to hold slightly differently formatted info.

```suggestion
        const numActiveColumns = selectedColumns.length;
```